File: app/src/main/java/com/example/acer/movies/adapters/RecyclerViewAdapterMovieCast.java
package com.example.acer.movies.adapters;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.acer.movies.models.Cast;
import com.example.acer.movies.R;
import com.example.acer.movies.network.URLConstants;
import com.squareup.picasso.Picasso;

import java.util.ArrayList;

/**
 * Created by KeshavAggarwal on 10/02/17.
 */

public class RecyclerViewAdapterMovieCast extends RecyclerView.Adapter<RecyclerViewAdapterMovieCast.ViewHolder> {
    private ArrayList<Cast> mMovieCast;
    Context mContext;

    public RecyclerViewAdapterMovieCast(ArrayList<Cast> movieCastMain, Context context) {
        mMovieCast = movieCastMain;
        mContext = context;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.movie_cast_list_item_view, parent, false);
        return new ViewHolder(v);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        if (mMovieCast != null) {
            holder.movieCastName.setText(mMovieCast.get(position).getName());
            Picasso.with(mContext).load(URLConstants.IMAGE_BASE_URL + mMovieCast.get(position).getProfile_path()).into(holder.movieCastImage);
            String character = "as " + mMovieCast.get(position).getCharacter();
            holder.movieCastCharacter.setText(character);
        }

    }

    @Override
    public int getItemCount() {
        return mMovieCast.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        ImageView movieCastImage;
        TextView movieCastName;
        TextView movieCastCharacter;
        View movieCastDivider;

        ViewHolder(View itemView) {
            super(itemView);
            movieCastImage = (ImageView) itemView.findViewById(R.id.movieCastImage);
            movieCastName = (TextView) itemView.findViewById(R.id.movieCastName);
            movieCastCharacter = (TextView) itemView.findViewById(R.id.moviCastCharacter);
            movieCastDivider = (View) itemView.findViewById(R.id.movieCastDivider);

        }
    }
}

File: app/src/main/java/com/app/techvalley/movies/adapters/RecyclerViewAdapterMovieCast.kt
package com.app.techvalley.movies.adapters

import android.content.Context
import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.app.techvalley.movies.models.Cast
import com.app.techvalley.movies.R
import com.app.techvalley.movies.network.URLConstants
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.movie_cast_list_item_view.view.*

import java.util.ArrayList

/**
 * Created by KeshavAggarwal on 10/02/17.
 */

class RecyclerViewAdapterMovieCast(private val mMovieCast: ArrayList<Cast>?, internal var mContext: Context) : RecyclerView.Adapter<RecyclerViewAdapterMovieCast.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.movie_cast_list_item_view, parent, false)
        return ViewHolder(v)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        if (mMovieCast != null) {
            holder.itemView.nameTV.text = mMovieCast[position].name
            Picasso.get().load(URLConstants.IMAGE_BASE_URL + mMovieCast[position].profile_path).into(holder.itemView.movieCastIV)
            val character = "as " + mMovieCast[position].character
            holder.itemView.castCharacterTV.text = character
        }

    }

    override fun getItemCount(): Int {
        return mMovieCast!!.size
    }

    class ViewHolder constructor(itemView: View) : RecyclerView.ViewHolder(itemView)
}
